### PR TITLE
add request_start_time variable

### PR DESCRIPTION
**Pull request by drawing**
add request_start_time variable

### CODE REVIEW

**Comment by chobits** [comment]
ACK and we need to introduce one test case for your new variables

File: src/http/ngx_http_variables.c [hunk 1]
```diff
@@ -1381,6 +1392,42 @@ ngx_http_variable_tcpinfo(ngx_http_request_t *r, ngx_http_variable_value_t *v,
...
+static char  *months[] = { "Jan", "Feb", "Mar", "Apr", "May", "Jun",
+                           "Jul", "Aug", "Sep", "Oct", "Nov", "Dec" };
+
+static ngx_int_t
+ngx_http_variable_request_start_time(ngx_http_request_t *r,
+    ngx_http_variable_value_t *v, uintptr_t data)
+{
+    u_char          *p;
+    ngx_time_t      *tp;
+    ngx_tm_t tm;
```
Comment (lhanjian) [inline src/http/ngx_http_variables.c:1405]: Can the number of spaces be unified here?

Comment (drawing) [reply src/http/ngx_http_variables.c]: There was indeed an issue with the alignment, I fixed it

**Comment by drawing** [comment]
> ACK and we need to introduce one test case for your new variables

Test case have been added. And the official website documentation has been updated: https://github.com/alibaba/tengine-website-generator/pull/42

**Review by chobits — APPROVED** [review]

**Comment by lhanjian** [comment]
LGTM

**Review by lhanjian — APPROVED** [review]

**Comment by lhanjian** [comment]
An additional question (just curious): under what observability requirements would you use this timing variable?

@drawing

**Comment by drawing** [comment]
> An additional question (just curious): under what observability requirements would you use this timing variable?
> 
> @drawing

For example, we want to count the QPS of Taobao's Double 11 promotion. Since the backend business processing time is different, many businesses even timeout and return after several seconds. $time_local is the time when the log is printed, the printing time will behind the request arrival time. The peak value The requests will be scattered and printed in the next few seconds. It is impossible to accurately count the number of peak requests per second. Using $request_start_time to count the peak values can solve this problem.

**Pull request merged**